tests/functional/src/main/java/org/mule/functional/junit4/ArtifactFunctionalTestCase.java

```diff
@@ -78,7 +79,12 @@ private static final void setPluginClassLoaders(List<ArtifactClassLoader> artifa
         {
             throw new IllegalStateException("Plugin class loaders were already set, it cannot be set again");
         }
-        pluginClassLoaders = artifactClassLoaders;
+
+        pluginClassLoaders = new ArrayList<>();
+        for (Object artifactClassLoader : artifactClassLoaders)
```

Comment:
I'll revert this change but still the thing about not casting to an ArtifactClassLoader is that instance was created by the runner classification process with the launcher classloader and at this point the TestClass was loaded using the isolated classloader so the cast won't work and I have to treat this object using reflection.